src/main.rs

```diff
@@ -264,10 +267,52 @@ fn create(command: args::CreateCommand) -> Result<(), Error> {
     }
 
     for script in presets.scripts {
+        if let Some(shared_dirs) = &script.shared_dirs {
+            for dir in shared_dirs {
+                // Copy shared directories to /tmp since bind mount is not read-only
```

Comment:
I don't think we should copy but instead bind-mount it directly, even if bind-mount can be made read-only. It's up to the user to make sure that his installation scripts doesn't cause any damage in the host directory

Reply:
I did this at first, but the issue is any error in the script causes the mounted directory to be deleted (when the process is killed and the loop device detached), deleting all files in the bind mount (even stuff like passwd failing because the two passwords you entered didn't match).

I couldn't get the remount as read-only option to work either, since we can't guarantee there aren't busy files on the source device.

The only problem I see with the copy is if the files are very large.

Reply:
You should use the [mount stack](https://github.com/r-darwish/alma/blob/master/src/storage/mount_stack.rs) to perform the mount. It will take care about un-mounting for you

Reply:
Nice, I'll try that.